media_driver/linux/common/os/mos_utilities_specific.c:2617:12: error: cast from pointer to smaller type 'uint32_t' (aka 'unsigned int') loses information
    return (uint32_t)pthread_self();
           ^~~~~~~~~~~~~~~~~~~~~~~~
media_driver/media_driver_next/linux/common/os/mos_utilities_specific_next.cpp:1761:12: error: cast from pointer to smaller type 'uint32_t' (aka 'unsigned int') loses information
    return (uint32_t)pthread_self();
           ^~~~~~~~~~~~~~~~~~~~~~~~

--- media_driver/linux/common/os/mos_utilities_specific.c.orig	2018-10-01 21:39:52 UTC
+++ media_driver/linux/common/os/mos_utilities_specific.c
@@ -2614,7 +2614,7 @@ uint32_t MOS_GetThreadId(
 
 uint32_t MOS_GetCurrentThreadId()
 {
-    return (uint32_t)pthread_self();
+    return (uintptr_t)pthread_self();
 }
 
 MOS_STATUS MOS_WaitThread(
--- media_driver/media_driver_next/linux/common/os/mos_utilities_specific_next.cpp.orig	2019-08-29 07:26:40 UTC
+++ media_driver/media_driver_next/linux/common/os/mos_utilities_specific_next.cpp
@@ -1758,7 +1758,7 @@ uint32_t MosUtilities::MOS_GetThreadId(
 
 uint32_t MosUtilities::MOS_GetCurrentThreadId()
 {
-    return (uint32_t)pthread_self();
+    return (uintptr_t)pthread_self();
 }
 
 MOS_STATUS MosUtilities::MOS_WaitThread(
